### PR TITLE
sc-66351 auto onboard large runenrs

### PR DESCRIPTION
**Pull request by rashidnhm**
**Context:**
This PR fixes an issue introduced as part of the previous pr #769 .

This issue was identified while trying to replicate an RC branch and testing the workflows as part of #773.

**Description of the Change:**
- Added extra echo statements to see clearly the outcome of the workflow if logic
- Removes quotes around the regex expression as that was making all evaluations of the logic to go false.
    - This issue did not arise in my previous testing but turned out to be a mismatch in shell configuration between myself locally and github actions. I was able to reproduce the bug locally and see that this worked.

#773 worked as expected with the changes introduced in this PR.

Links:
- https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/9616650952/job/26526641408?pr=773#step:2:14
- https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/9616650967/job/26526641639?pr=773#step:2:14

**Benefits:**
Auto on-boarding tested and working with this fix.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — COMMENTED** [review]
Could you please re-open your test PR and rebase it to this PR's branch?

**Review by AmintorDusko — APPROVED** [review]
Thank you very much!

File: .github/workflows/determine-workflow-runner.yml [hunk 1]
```diff
@@ -41,6 +41,9 @@ jobs:
           RC_BRANCH_FORMAT_REGEX: v[0-9]+\.[0-9]+\.[0-9]+-rc[0-9]?
           REPO_FULL_NAME: PennyLaneAI/pennylane-lightning
         run: |
-          if [[ '${{ contains(github.event.pull_request.labels.*.name, 'urgent') }}' == 'true' || ('${{ github.event.pull_request.head.repo.full_name }}' == "$REPO_FULL_NAME" && "${{ github.event.pull_request.head.ref }}" =~ "$RC_BRANCH_FORMAT_REGEX") ]]; then
+          if [[ '${{ contains(github.event.pull_request.labels.*.name, 'urgent') }}' == 'true' || ('${{ github.event.pull_request.head.repo.full_name }}' == "$REPO_FULL_NAME" && "${{ github.event.pull_request.head.ref }}" =~ $RC_BRANCH_FORMAT_REGEX) ]]; then
```
Comment (maliasadi) [inline .github/workflows/determine-workflow-runner.yml:44]:
```suggestion
          if [[ '${{ contains(github.event.pull_request.labels.*.name, 'urgent') }}' == 'true' || ('${{ github.event.pull_request.head.repo.full_name }}' == $REPO_FULL_NAME && '${{ github.event.pull_request.head.ref }}' =~ $RC_BRANCH_FORMAT_REGEX) ]]; then
```

**Review by maliasadi — COMMENTED** [review]
LGTM! Don't forget to update the changelog:

``` markdown
* Release candidate branches are automatically switched over to using the new large GitHub runner pool.
  [(#769)](https://github.com/PennyLaneAI/pennylane-lightning/pull/769)
  [(#774)](https://github.com/PennyLaneAI/pennylane-lightning/pull/774)
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -98,6 +98,9 @@
 
 ### Bug fixes
 
+* Fix regex matching issue with auto on-boarding of release candidate branch to using the large runner queue
```
Comment (maliasadi) [inline .github/CHANGELOG.md:101]:
```suggestion
* Fix regex matching issue with auto on-boarding of release candidate branch to using the large runner queue.
```

**Review by maliasadi — APPROVED** [review]
Thanks @rashidnhm! 🙌

**Pull request merged**